web/src/peer_data.ts

```diff
@@ -317,11 +317,19 @@ export async function maybe_fetch_is_user_subscribed(
     return subscribers.has(user_id);
 }
 
-export function get_unique_subscriber_count_for_streams(stream_ids: number[]): number {
+export async function get_unique_subscriber_count_for_streams(
+    stream_ids: number[],
+): Promise<number> {
     const valid_subscribers = new LazySet([]);
+    const promises: Record<number, Promise<LazySet | null>> = {};
+    for (const stream_id of stream_ids) {
+        promises[stream_id] = get_full_subscriber_set(stream_id, false);
+    }
 
     for (const stream_id of stream_ids) {
-        const subscribers = get_loaded_subscriber_subset(stream_id);
+        // If it's `null`, that means a request failed and we don't know the
+        // full subscribers set, so just use whatever we have already.
```

Comment:
I'm not sure this is the right move, since we'd be showing inaccurate data. Maybe we actually want to just remove the message if we don't know the count?

Reply:
Yeah, we should never shown wrong data. I think options include:
- Show "At least" at the cost of extra strings.
- Just show a loading indicator until we have the data.

I'd be surprised if in practice it were common to have the loading indicator experience do anything but succeed after <5 seconds -- this is not an endpoint where we expect errors if you're on the Internet. So I think probably that second experience is better.

Reply:
Updated with a loading spinner!